lib/bootloader_setup.pm

```diff
@@ -1457,18 +1461,20 @@ C<$search> if set, bypass default grub cmdline variable.
 =cut
 
 sub add_grub_cmdline_settings {
+    my $search_variable = (is_bootloader_grub2) ? GRUB_CMDLINE_VAR : '';
+    my $needs_quote = (is_bootloader_grub2) ? '"' : '';
     my $add = shift;
     my %args = testapi::compat_args(
         {
             add => $add,
             update_grub => 0,
-            search => '^' . GRUB_CMDLINE_VAR,
+            search => '^' . $search_variable,
         },
         ['update_grub', 'search'],
         @_
     );
 
-    change_grub_config('"$', " $add\"", $args{search}, "g", $args{update_grub});
+    change_grub_config($needs_quote . '$', " " . $add . $needs_quote, $args{search}, "g", $args{update_grub});
```

Comment:
This probably could be changed to (same for `add_grub_cmdline_settings` => `add_bootloader_cmdline_settings`
```suggestion
    change_bootloader_config($needs_quote . '$', " " . $add . $needs_quote, $args{search}, "g", $args{update_grub});
```